bfasst/utils/structural.py

```diff
@@ -277,47 +323,87 @@ def map_ports(self):
+            for instance in all_instances:
+                properties = set()
+                for prop in self.get_properties_for_type(instance.cell_type):
+                    properties.add(
+                        f"{prop}{convert_verilog_literal_to_int(instance.properties[prop])}"
+                    )
+                    # properties[prop] = convert_verilog_literal_to_int(instance.properties[prop])
+
+                grouped_by_cell_type[(instance.cell_type, hash(frozenset(properties)))].append(
+                    instance
                 )
-            my_hash = hash(frozenset(properties))
 
-            instances_matching = grouped_by_cell_type[(named_instance.cell_type, my_hash)]
+            for named_instance in self.named_netlist.instances_to_map:
+                ###############################################################
+                # First find all instances of the same type and same properties
+                ###############################################################
 
-            if not instances_matching:
-                logging.info(
-                    "No property matches for cell %s of type %s. Properties:",
-                    named_instance.name,
-                    named_instance.cell_type,
-                )
+                # Compute a hash of this instance's properties
+                properties = set()
                 for prop in self.get_properties_for_type(named_instance.cell_type):
-                    logging.info("  %s: %s", prop, named_instance.properties[prop])
-                raise StructuralCompareError(
-                    f"Not equivalent. {named_instance.name} has no possible match in the netlist."
-                )
+                    properties.add(
+                        f"{prop}{convert_verilog_literal_to_int(named_instance.properties[prop])}"
+                    )
+                my_hash = hash(frozenset(properties))
+
+                instances_matching = grouped_by_cell_type[(named_instance.cell_type, my_hash)]
+
+                if not instances_matching:
+                    logging.info(
+                        "No property matches for cell %s of type %s. Properties:",
+                        named_instance.name,
+                        named_instance.cell_type,
+                    )
+                    for prop in self.get_properties_for_type(named_instance.cell_type):
+                        logging.info("  %s: %s", prop, named_instance.properties[prop])
+                    raise StructuralCompareError(
+                        f"Not equivalent. {named_instance.name} \
+                        has no possible match in the netlist."
+                    )
 
-            self.possible_matches[named_instance] = instances_matching.copy()
+                self.possible_matches[named_instance] = instances_matching.copy()
+
+            possible_matches = {}
+            # pylint: disable=consider-using-dict-items
```

Comment:
Why did you decide to disable this check? Maybe I'm missing why dict items wasn't quite working.